src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -936,6 +962,23 @@ private void SetSessionHandleOptions(SafeWinHttpHandle sessionHandle)
             SetSessionHandleTlsOptions(sessionHandle);
             SetSessionHandleTimeoutOptions(sessionHandle);
             SetDisableHttp2StreamQueue(sessionHandle);
+            SetTcpKeepalive(sessionHandle);
+        }
+
+        private unsafe void SetTcpKeepalive(SafeWinHttpHandle sessionHandle)
+        {
+            if (TcpKeepAliveEnabled)
+            {
+                var tcpKeepalive = new Interop.WinHttp.tcp_keepalive
+                {
+                    onoff = 1,
+                    keepaliveinterval = (uint)_tcpKeepAliveInterval.TotalMilliseconds,
+                    keepalivetime = (uint)_tcpKeepAliveTime.TotalMilliseconds
+                };
+                void* ptr = &tcpKeepalive;
```

Comment:
Nit: this local seems unnecessary, and you can just use `(IntPtr)&tcpKeepalive` below.